collect_app/src/main/java/org/odk/collect/android/widgets/utilities/DateTimeWidgetUtils.java

```diff
@@ -0,0 +1,124 @@
+package org.odk.collect.android.widgets.utilities;
+
+import android.content.Context;
+import android.os.Build;
+import android.os.Bundle;
+import android.widget.TextView;
+
+import org.javarosa.core.model.data.TimeData;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.joda.time.DateTime;
+import org.joda.time.LocalDateTime;
+import org.odk.collect.android.activities.FormEntryActivity;
+import org.odk.collect.android.fragments.dialogs.BikramSambatDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.CopticDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.CustomDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.CustomTimePickerDialog;
+import org.odk.collect.android.fragments.dialogs.EthiopianDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.FixedDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.IslamicDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.MyanmarDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.PersianDatePickerDialog;
+import org.odk.collect.android.logic.DatePickerDetails;
+import org.odk.collect.android.utilities.DateTimeUtils;
+import org.odk.collect.android.utilities.DialogUtils;
+import org.odk.collect.android.utilities.ThemeUtils;
+
+import java.util.Date;
+
+import static org.odk.collect.android.fragments.dialogs.CustomDatePickerDialog.DATE_PICKER_DIALOG;
+import static org.odk.collect.android.fragments.dialogs.CustomTimePickerDialog.CURRENT_TIME;
+import static org.odk.collect.android.fragments.dialogs.CustomTimePickerDialog.TIME_PICKER_THEME;
+import static org.odk.collect.android.fragments.dialogs.FixedDatePickerDialog.CURRENT_DATE;
+import static org.odk.collect.android.fragments.dialogs.FixedDatePickerDialog.DATE_PICKER_DETAILS;
+import static org.odk.collect.android.fragments.dialogs.FixedDatePickerDialog.DATE_PICKER_THEME;
+
+public class DateTimeWidgetUtils {
+
+    private DateTimeWidgetUtils() {
+    }
+
+    public static TimeData getTimeData(int hourOfDay, int minuteOfHour, boolean nullAnswer) {
+        // use picker time, convert to today's date, store as utc
+        DateTime localDateTime = new DateTime()
+                .withTime(hourOfDay, minuteOfHour, 0, 0);
+
+        return !nullAnswer
+                ? new TimeData(localDateTime.toDate())
+                : null;
+    }
+
+    public static void setDateLabel(Context context, TextView dateAnswerText, Date date, DatePickerDetails datePickerDetails) {
```

Comment:
it should be a part of widget class, passing `TextView dateAnswerText` just to setText on it is strange.